src/openzaak/components/catalogi/api/validators.py

```diff
@@ -443,3 +445,48 @@ def __call__(self, attrs):
                             f"Objects can't be updated with a relation to non-concept {field_name}"
                         )
                         raise ValidationError(msg, code=self.code)
+
+
+class GeldigheidVersiedatumValidator:
+    """
+    De begin_datum is gelijk aan een Versiedatum van het gerelateerde zaaktype.
+
+    De datum_einde_geldigheid is gelijk aan of gelegen na de datum zoals opgenomen
+    onder 'Datum begin geldigheid resultaattype’.
+    De datum is gelijk aan de dag voor een Versiedatum van het gerelateerde zaaktype.
+
+    This validation is used only in API since in admin Versiedatum is filled automatically
+    """
+
+    code = "versiedatum-mismatch"
+
+    def set_context(self, serializer):
+        # Determine the existing instance, if this is an update operation.
+        self.instance = getattr(serializer, "instance", None)
+
+    def __call__(self, attrs):
+        datum_begin_geldigheid = (
+            attrs.get("datum_begin_geldigheid") or self.instance.datum_begin_geldigheid
+        )
+        versiedatum = attrs.get("versiedatum") or self.instance.versiedatum
+
+        if datum_begin_geldigheid != versiedatum:
+            msg = _(
+                "De datum_begin_geldigheid moet gelijk zijn aan een "
+                "Versiedatum van het gerelateerde zaaktype."
+            )
+            raise ValidationError(msg, code=self.code)
+
+        # current_einde_geldigheid = (
```

Comment:
@sergei-maertens @joeribekker 
I've copied logic here from `GeldigheidMixin._clean_geldigheid` because I removed this check from `ZaakType` model
I've commented this part because it seems to contradict the code above:
`versiedatum` cant be equal to `datum_begin_geldigheid` and to `datum_einde_geldigheid + 1` at the same time. 

Is it correct? Should I uncomment it? Should I comment it in the `GeldigheidMixin._clean_geldigheid`?

Reply:
I think that commented code can be removed. It is possible that version x exists from `[a,b]` and version y exists from `[c,d]`, with a gap between `b` and `c`. In practice, that means that between b & c no cases of that type can be created.